Content.Client/CrewMedal/UI/CrewMedalWindow.xaml.cs

```diff
@@ -0,0 +1,66 @@
+using Robust.Client.UserInterface.CustomControls;
+using Robust.Client.AutoGenerated;
+using Robust.Client.UserInterface.XAML;
+
+namespace Content.Client.CrewMedal.UI;
+
+[GenerateTypedNameReferences]
+public sealed partial class CrewMedalWindow : DefaultWindow
+{
+    public event Action<string>? OnReasonChanged;
+
+    /// <summary>
+    /// Is the user currently entering text into the control?
+    /// </summary>
+    private bool _focused;
+
+    private string _reason = string.Empty;
+    private bool _awarded = false;
+
+    private int _maxCharacters = 50;
+
+    public CrewMedalWindow()
+    {
+        RobustXamlLoader.Load(this);
+
+        ReasonLineEdit.OnTextChanged += _ =>
+        {
+            SaveButton.Disabled = _awarded || ReasonLineEdit.Text.Length > _maxCharacters;
+            CharacterLabel.Text = Loc.GetString("crew-medal-ui-character-limit", ("number", ReasonLineEdit.Text.Length), ("max", _maxCharacters));
+        };
+        ReasonLineEdit.OnFocusEnter += _ => _focused = true;
+        ReasonLineEdit.OnFocusExit += _ => _focused = false;
+
+        SaveButton.OnPressed += _ =>
+        {
+            OnReasonChanged?.Invoke(ReasonLineEdit.Text);
+            SaveButton.Disabled = true;
+        };
+
+        CharacterLabel.Text = Loc.GetString("crew-medal-ui-character-limit", ("number", ReasonLineEdit.Text.Length), ("max", _maxCharacters));
+    }
+
+    public void SetCurrentReason(string reason)
+    {
+        if (_reason == reason)
+            return;
+
+        _reason = reason;
+        if (!_focused)
```

Comment:
Partly for my own curiosity, but do you know under what conditions Reload() can be called while the user already has the window open? 

Could you use ReasonLineEdit.HasKeyboardFocus() here? Would allow you to remove the _focused and the pair of signal subscriptions.